io/disk/disktest.py

```diff
@@ -0,0 +1,115 @@
+        if not softm.check_installed("gcc") and not softm.install("gcc"):
+            self.error('Gcc is needed for the test to be run')
+        self.disks = self.params.get('disks', default=None)
+        self.gigabytes = self.params.get('gigabytes', default=None)
+        self.chunk_mb = self.params.get('chunk_mb', default=None)
+        self.memory_mb = memory.memtotal() / 1024 / 8
+        self.data_dir = os.path.join(os.path.abspath(self.datadir))
+        self.disk_srcdir = os.path.join(os.path.abspath(self.datadir), 'src')
+        self.build(self.disk_srcdir)
+
+    def build(self, srcdir):
+        """
+        Compiles disktest.
+        """
+        os.chdir(srcdir)
+        build.make(srcdir, extra_args='clean')
+        build.make(srcdir)
+
+    def one_disk_chunk(self, disks, chunk):
+        """
+        Tests one part of the disk by spawning a disktest instance.
+        :param disks: Directory (usually a mountpoint).
+        :param chunk: Portion of the disk used.
+        """
+        logging.info("Testing %d MB files on %s in %d MB memory, chunk %s",
+                     self.chunk_mb, disks, self.memory_mb, chunk)
+        logfile = self.outputdir + '/log.txt'
+        cmd = ("%s/disktest -m %d -f %s/testfile.%d -i -S >>%s 2>&1" %
+               (self.disk_srcdir, self.chunk_mb, disks, chunk, logfile))
+        logging.debug("Running '%s'", cmd)
+        thread = subprocess.Popen(cmd, shell=True)
+        return thread.pid
+
+    def cleanup(self, disks):
+        """
+        To clean all the testfiles generated
+        """
+        process.run('rm -rf %s/testfile.*' % disks, shell=True)
+
+    def test(self):
+        """
+        Runs one iteration of disktest.
+        :param disks: Directory (usually mountpoints) to be passed
+                to the test. Directory gets created if not existing
+        :param gigabytes: Disk space that will be used for the test to run.
+        :param chunk_mb: Size of the portion of the disk used to run the test.
+                Cannot be larger than the total amount of free RAM.
+        """
+        os.chdir(self.disk_srcdir)
+        if self.chunk_mb is None:
+            self.chunk_mb = memory.memtotal() / 1024 / 8
+        if self.disks is None:
+            self.disks = self.data_dir
+        if self.gigabytes is None:
+            free = 100  # cap it at 100GB by default
+            if not os.path.isdir(self.disks):
+                os.makedirs(self.disks)
+
+            free = min(disk.freespace(self.disks) / 1024 ** 3, free)
+            self.gigabytes = free
+            logging.info("Resizing to %s GB", self.gigabytes)
+            sys.stdout.flush()
+
+        if self.memory_mb > self.chunk_mb:
+            self.error("Too much RAM (%dMB) for this test to work" %
+                       self.memory_mb)
+
+        chunks = (1024 * self.gigabytes) / self.chunk_mb
+        logging.info("Total of disk chunks that will be used: %s", chunks)
```

Comment:
@Nosheen08 up to this point it's still the initialization. So please move the above to the `setUp` part and use `self.skip` to mark test as `SKIP` when it does not match the specification (instead of `error`).